src/components/RequestInvite.js

```diff
@@ -0,0 +1,200 @@
+        </div>
+      </div>
+    );
+  }
+
+  renderInviteInstruction() {
+    const myInviteUrl = `${window.location.origin}/m/${this.props.authMemberDoc.get('mid')}/invite`;
+    return (
+      <div>
+        <div>
+          Hi there {this.props.authMemberDoc.get('full_name')}!
+        </div>
+        <div>
+          If you would like to invite someone else, make a video where each of you say your full name and (optionally)
+          why you want to join Raha, have them upload it to Youtube, then direct them to
+          visit <a href={myInviteUrl}>{myInviteUrl}</a> to upload their own video and make their account. After they do this,
+          you will have to go visit their profile, check the video is accurate, and hit "Trust" for them to become members.
+          Please bear with us as we work on making this process more conveniant, if you have any thoughts reach us at
+          <a href="mailto:ideas@raha.io?subject=Raha%20Improvement">ideas@raha.io</a> or file a <a href="https://github.com/rahafoundation/raha.io/issues">Github Issue</a>.
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    if (this.state.submitted) {
+      return (
+        <div>
+          Your video has been submitted for review! After approval by us
+          and {this.props.toMemberDoc.get('full_name')}, your profile will appear at
+          to <a href="https://raha.io/me">https://raha.io/me</a>.
+          We are available at <a href="mailto:help@raha.io">help@raha.io</a> if you have any questions.
+        </div>
+      );
+    }
+    if (!this.props.notSignedIn && !this.props.isAuthMemberDocLoaded) {
+      // TODO the loading appears again breifly is user goes from logged out to signed in with this.renderLogIn()
+      return <div>Loading</div>;
+    }
+    if (this.props.authMemberDoc && this.props.authMemberDoc.get('invite_confirmed')) {
+      return this.renderInviteInstructions();
+    }
+    return (
+      <div>
+        <b>Request Invite</b>
+        {this.renderDirections()}
+        {this.props.notSignedIn ? this.renderLogIn() : (this.props.isAuthLoaded ? this.renderForm() : <div>Loading</div>)}
+      </div >
+    );
+  }
+}
+
+function mapStateToProps(state, ownProps) {
+  const memberId = ownProps.match.params.memberId;
+  const isAuthLoaded = state.auth.isLoaded;
+  if (!isAuthLoaded) {
+    return { isAuthLoaded, memberId };
+  }
+  const authFirebaseUser = state.auth.firebaseUser;
+  if (!authFirebaseUser) {
+    return { isAuthLoaded, memberId, notSignedIn: true };
+  }
+  const authMember = state.members.byUid[authFirebaseUser.uid];
+  return {
+    isAuthLoaded,
+    authFirebaseUser,
+    memberId,
+    isAuthMemberDocLoaded: authMember && !authMember.isFetching,
+    authMemberDoc: getAuthMemberDoc(state),
+    fullName: state.auth.firebaseUser.displayName,
+    toMemberDoc: getMemberDocByMid(state, ownProps.match.params.memberId)
```

Comment:
could we use `memberId` here instead? e.g. `getMemberDocByMid(state, memberId)` or do they refer to different things?

Reply:
The memberDoc is firebase doc where we can do `.get('full_name')` etc ... we should get Flow actually working and be better about documenting our types (perhaps with a linter), we've probably now gone a bit too far from "full typescript" to "barely typed"